UserTools/LoadWCSim/LoadWCSim.cpp

```diff
@@ -631,18 +636,24 @@ bool LoadWCSim::Execute(){
 			} else {
 				// instead take the true time of the first photon
 				std::vector<int> photonids = digihit->GetPhotonIds();   // indices of the digit's photons
+				std::cout << "LoadWCSim Tool: number of photons in this hit: " << photonids.size() << endl;
 				double earliestphotontruetime=999999999999;
-				for(int& aphotonindex : photonids){
-					WCSimRootCherenkovHitTime* thehittimeobject =
-						 (WCSimRootCherenkovHitTime*)firsttrigt->GetCherenkovHitTimes()->At(aphotonindex);
-					if(thehittimeobject==nullptr){
-						cerr<<"LoadWCSim Tool: ERROR! Retrieval of photon from digit returned nullptr!"<<endl;
-						continue;
+				if (firstHit) {
+					for (int& aphotonindex : photonids) {
+						WCSimRootCherenkovHitTime* thehittimeobject =
+							(WCSimRootCherenkovHitTime*)firsttrigt->GetCherenkovHitTimes()->At(aphotonindex);
+						if (thehittimeobject == nullptr) {
+							cerr << "LoadWCSim Tool: ERROR! Retrieval of photon from digit returned nullptr!" << endl;
+							continue;
+						}
+						double aphotontime = static_cast<double>(thehittimeobject->GetTruetime());
+						if (aphotontime < earliestphotontruetime) { earliestphotontruetime = aphotontime; }
 					}
-					double aphotontime = static_cast<double>(thehittimeobject->GetTruetime());
-					if(aphotontime<earliestphotontruetime){ earliestphotontruetime = aphotontime; }
+					digittime = earliestphotontruetime;
+				}
+				else {
+
```

Comment:
...? Is there supposed to be some alternative here? Otherwise if `firstHit` is false, we just leave the digit times uninitialised. :raised_eyebrow: